process-document/src/main/kotlin/com/ritense/processdocument/service/DocumentDelegateService.kt

```diff
@@ -0,0 +1,122 @@
+) {
+
+    private val mapper = Mapper.INSTANCE.get()
+
+    fun getDocumentVersion(execution: DelegateExecution): DocumentVersion? {
+        logger.debug("Get version of document {}", execution.processBusinessKey)
+        return getDocumentById(execution).version()
+    }
+
+    fun getDocumentCreatedOn(execution: DelegateExecution): LocalDateTime? {
+        logger.debug("Get created on date of document {}", execution.processBusinessKey)
+        return getDocumentById(execution).createdOn()
+    }
+
+    fun getDocumentCreatedBy(execution: DelegateExecution): String? {
+        logger.debug("Get created by of document {}", execution.processBusinessKey)
+        return getDocumentById(execution).createdBy()
+    }
+
+    fun getDocumentModifiedOn(execution: DelegateExecution): LocalDateTime? {
+        logger.debug("Get modified on of document {}", execution.processBusinessKey)
+        return getDocumentById(execution).modifiedOn().getOrNull()
+    }
+
+    fun getDocumentAssigneeId(execution: DelegateExecution): String? {
+        logger.debug("Get assigneeId of document {}", execution.processBusinessKey)
+        return getDocumentById(execution).assigneeId()
+    }
+
+    fun getDocumentAssigneeFullName(execution: DelegateExecution): String? {
+        logger.debug("Get assignee full name of document {}", execution.processBusinessKey)
+        return getDocumentById(execution).assigneeFullName()
+    }
+
+    fun getDocumentById(execution: DelegateExecution): Document {
+        val documentId = processDocumentService.getDocumentId(CamundaProcessInstanceId(execution.processInstanceId), execution)
+        return jsonSchemaDocumentService.getDocumentBy(documentId)
+    }
+
+    fun findValueByJsonPointer(jsonPointer: String?, execution: DelegateExecution?): Any {
+        return findOptionalValueByJsonPointer(jsonPointer, execution!!).orElseThrow()
+    }
+
+    fun findValueByJsonPointerOrDefault(jsonPointer: String?, execution: DelegateExecution, defaultValue: Any): Any {
+        return findOptionalValueByJsonPointer(jsonPointer, execution).orElse(defaultValue)
+    }
+
+    private fun findOptionalValueByJsonPointer(jsonPointer: String?, execution: DelegateExecution): Optional<Any> {
+        val jsonSchemaDocumentId = JsonSchemaDocumentId.existingId(UUID.fromString(execution.processBusinessKey))
+        logger.debug("Retrieving value for key {} from documentId {}", jsonPointer, execution.processBusinessKey)
+        return AuthorizationContext.runWithoutAuthorization {
+            documentService.findBy(jsonSchemaDocumentId)
+        }.flatMap { jsonSchemaDocument ->
+            jsonSchemaDocument.content().getValueBy(JsonPointer.valueOf(jsonPointer))
+        }
+            .map(::transform)
+    }
+
+    private fun transform(jsonNode: JsonNode): Any? {
+        if (jsonNode.isNumber) {
+            // Removing this would result in a breaking change, as 3.0 will become an int when using treeToValue
```

Comment:
@marijnritense Should we maybe make this breaking change now by removing the number/double check?

Reply:
I think this should be at least delayed until 12. This can end up resulting in quite a bit more work for implementation teams.